### ISSUE TITLE
Quick capture mobile app

### ISSUE THREAD

**Issue by davidpdrsn** [davidpdrsn/CardistryIO#227]
Once https://github.com/davidpdrsn/CardistryIO/issues/24 is done I would like to get our feet wet with making a mobile app.

I think a good place to start is an app that allows you quickly record and upload content to the site. That way you wouldn't have to first transfer it to your computer and then upload it.

**Comment by bendobos** [comment]
Absolutely! As I said before I'd really love to take the hybrid approach and use [Ionic](http://ionicframework.com/) to build platform independently. I think we could even wire that up in a way that it works nicely with rails if you would prefer that. However, Ionic is awesome and if we could expose the data in a REST API and have the mobile page done, we could have iOS, Android and even Windows Phone apps that look and feel nice in a weekend. Another plus is that we could maintain the app together because I have never worked with native Swift or Java apps. We'd also save a lot of time and work by building one app and not two.

**Comment by davidpdrsn** [comment]
This thing I an proposing would probably be somewhere around three screens.
1. Login
2. Capture video/photo
3. Add name, description, credits, etc.

Thats it. For something as simple as this I can't see why we wouldn't go fully native. We might need to have full mobile app versions of the site, but let's worry about that when the web version is shipped and works well.

That being said I think it is very likely we will need mobile apps at some point where people can browse the feed, watch videos, etc, and while I do think hybrid makes the most sense, I don't Ionic is the way to go. I think it looks overly complicated and I don't like the thought of rebuilding the whole app as a REST API. I also don't like the extra complexity that Angular and a virtual DOM brings.

If I had to build mobile apps today I would go with Turbolinks 5 which is built by people from the Rails team. This seems like a much simpler solution because there is no client side MVC framework, and no virtual DOM. It is just response web views that are rendered on the server and sent to the client. You can see Turbolinks 5 in action and hear the motivation behind it [in this talk](https://www.youtube.com/watch?v=SWEts0rlezA).

**Comment by bendobos** [comment]
Ah, I misunderstood your title and was thinking of a full-fledged mobile app.

I didn't know about Turbolinks. Looks cool. Why is it not in the project yet? Even without mobile it'd be awesome to not have a page reload when switching routes.

**Comment by davidpdrsn** [comment]
I used a Rails template maintained by thoughbot which doesn't include turbolinks by default. I'll look into adding it back.
